src/Chapter3.hs

```diff
@@ -910,6 +1008,19 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+
+instance Append Gold where
+  append (Gold x) (Gold y) = Gold $ x + y
+
+instance Append [a] where
+  append x y = x ++ y
+
+instance (Append a) => Append (Maybe a) where
+  append (Just x) (Just y) = Just $ append x y
+  append (Just x) Nothing = Just x
+  append Nothing (Just y)= Just y
+  append Nothing Nothing = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂